java/src/com/github/bazel_contrib/contrib_rules_jvm/junit5/README.md

```diff
@@ -0,0 +1,280 @@
+test name (eg. `testMethod{String)[1]`).
+
+#### Pathological
+
+Creates a single test class with a top-level `testsuite` tag named `JUnit
+Jupiter`. This contains a single `testcase` tag with a `name` attribute 
+matching the method signature of the ignored test case (eg. `example(String)
+`) no matter how many tests would have been generated. The single `testcase` 
+contains a nested `skipped` tag, containing the reason for the test being 
+skipped.
+
+### Bazel's JUnit4 runner
+
+Obviously, not every test type supported by JUnit5 can be represented by 
+JUnit4, but for those that can....
+
+In all cases, Bazel generates a single file per test target. The generated 
+`testsuite` tags each have an empty `properties` tag. Bazel is also the only 
+test runner that generates a top-level `testsuites` tag for each file.
+
+#### Simple
+
+Below the `testsuites` tag is a single `testsuite` tag. This is named after 
+the test class name. Nested below this is a `testcase` for each test method. 
+The `name` attribute is the name of the test method without type information 
+(eg. `example`)
+
+#### Suite
+
+#### Dynamic
+
+Skipped
+
+#### Parameterised
+
+Parameterised tests work slightly differently in JUnit 4 than they do in 
+JUnit5. In JUnit4, each parameter is passed to the constructor of the test 
+class, meaning that each test class must store that value on a field for it 
+to be accessible to test methods.
+
+Below the `testsuites` tag is a `testsuite` tag for the test class itself. 
+This is basically empty.
+
+For each generated test case, there is a `testsuite` entry, named after the 
+order in which the suite was generated (eg. `[0]` or `[1]`). Within each of 
+those `testsuite` entries is a `testcase` per method in the test class, 
+where each `testcase` has a `name` attribute that is a combination of the 
+test method name without signature information and the run number (eg. 
+`someTest[1]`)
+
+#### Pathological
+
+Skipped
+
+## Desired behaviour
+
+The output generated by this runner has to meet the two requirements listed 
+at the start of the doc. This means that we shall always issue a top-level 
+`testsuites` tag. The XSD makes it clear that this can only contain 
+`testsuite` tags, and these can only contain `testcase` tags. 
+
+When executing tests, JUnit5 prepares a test plan. This contains all the 
+statically defined tests that have been discovered by each test engine that 
+is available to it. These are arranged in a hierarchy, with "test roots" 
+being the actual engine being used. These test roots are used by the legacy 
+XML listener to provide the suite name, which is why they bear no relation 
+to a class name at all. 
+
+Neither Gradle nor Maven expose the root containers, so it's appropriate 
+that we don't do that either.
+
+The reason that the legacy XML listener is outputting top-level test suites 
+with names like `JUnit Platform Suite` is that 
```

Comment:
What is the reason? *from the edge of seat* I gotta know! (probably something to do with JUnit 4 compatibility?)